src/iotjs_util.h

```diff
@@ -35,5 +35,10 @@ void iotjs_buffer_release(char* buff);
 #define IOTJS_RELEASE(ptr) /* Release memory allocated by IOTJS_ALLOC() */ \
   iotjs_buffer_release((char*)ptr)
 
+#define IOTJS_SAFE_RELEASE(ptr)       \
+  ({                                  \
+    iotjs_buffer_release((char*)ptr); \
+    ptr = NULL;                       \
```

Comment:
Can't we just simply move the `ptr=NULL` part into the buffer release method? Is there any benefit having it as a separate macro?

Reply:
Yes, we can move it into iotjs_buffer_release as you mentioned. There are two below options, and I prefer to use (b). Are you fine with (b)?

a) use double pointers
```
char *ptr;
iotjs_buffer_release(&ptr); 

void iotjs_buffer_release(char** buffer)
```

b) ptr passing by reference
```
iotjs_buffer_release(ptr); 

void iotjs_buffer_release(char* & buffer)
```

Reply:
IMHO, we can not pass by reference. We are in C and not C++. However now I understand why the new macro was created.

Reply:
@galpeter You're right. b) isn't possible in C. I forgot that.

I receive another opinion to modify IOTJS_RELEASE instead of adding a new macro. I will change this commit like that.

Reply:
I'm ok with updating the IOTJS_RELEAS macro.